improver/blending/weighted_blend.py

```diff
@@ -579,8 +579,8 @@ def non_percentile_weights(self, cube, weights):
                 np.float32
             )
         (blend_dim,) = cube.coord_dims(self.blend_coord)
-        self.check_weights(weights_array, blend_dim)
-        return weights_array.astype(np.float32)
+
+        return weights_array
```

Comment:
Could you make the same change in `percentile_weights`? I think the same applies there.